qt/main.cpp

```diff
@@ -215,20 +214,19 @@ int main(int argc, char * argv[])
     fmt.setSwapInterval(1);
     fmt.setDepthBufferSize(16);
 
-    // This is a workaround for some systems,
-    // including MacOs and older Linux distros, which rely on X.org and Mesa.
-    // Where somehow the driver itself doesn't
-    // make all the otherwise supported GLSL versions available by default
-    // and such requests are somehow disregarded at later stages of execution.
-    // This setting here will be potentially overwritten and overruled anyway,
-    // and only needed to ensure that we have the needed GLSL compiler available
-    // later when we need it.
-    if (app.platformName() == QString("xcb") ||
-        app.platformName() == QString("cocoa"))
+    if (app.platformName() == QString("wayland"))
+    {
+      // TODO: Check if there is a way to use OpenGL 3.2 on wayland platform.
+      fmt.setProfile(QSurfaceFormat::CompatibilityProfile);
+      fmt.setVersion(2, 1);
```

Comment:
@Ferenc- what's the reason for the failing Wayland? Does it support OpenGL 2.1?
Can you please check this branch with the version set to 2.0? And if it doesn't work, then with CoreProfile, and if it doesn't work, then try removing these two lines completely.

We need to get it working reliably, without guessing and magic.

Reply:
Do we rely on any 2.1 features? Can we force using 2.0 everywhere explicitly? Because we now explicitly request 2.1, while by default, without explicit specification, Qt requests 2.0 (and it works).

Reply:
> Do we rely on any 2.1 features? 

As far as I can tell, and as far as my tests go, nothing relied on it. But that does not expclude the possibility.

>  Can we force using 2.0 everywhere explicitly?

In the use cases that I am aware of, and I test, yes.